test/common/router/router_test.cc

```diff
@@ -868,6 +872,121 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
   EXPECT_TRUE(verifyHostUpstreamStats(1, 0));
 }
 
+// Validate that x-envoy-attempt-count is added to request headers when option is true.
+TEST_F(RouterTest, EnvoyAttemptCountInRequest) {
+  setIncludeAttemptCount(true);
+
+  EXPECT_CALL(cm_.conn_pool_, newStream(_, _)).WillOnce(Return(&cancellable_));
+  expectResponseTimerCreate();
+
+  Http::TestRequestHeaderMapImpl headers;
+  HttpTestUtility::addDefaultHeaders(headers);
+  router_.decodeHeaders(headers, true);
```

Comment:
Also optional but what do you think of setting up some utility functions to reduce cut and paste code here?